### DISCUSSION TITLE
How to get codespace usage core hours using github rest api??

### DISCUSSION THREAD

**Discussion by Railway-Hobby** [community/community#138087] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

How to get codespace usage core hours using github rest api??

**Comment by rahulkottikasala** [comment, marked as answer]
To get Codespace usage core hours using GitHub's REST API, you can make a GET request to the following endpoint:
GET /user/codespaces/usage
You'll need to authenticate with a token that has the codespaces scope.

**Reply by cschot** [reply]
Api endpoint `GET /user/codespaces/usage` does not exists anymore.

When trying to fetch used core hours you should use the `GET /users/USERNAME/settings/billing/usage` from now on.
This endpoint was only available to users with access to the enhanced billing platform.
Which made it impossible to use for users with personal accounts,
But GitHub is rolling out the enhanced billing platform to everyone at the moment. 

See the changelog https://github.blog/changelog/2025-02-25-enhanced-billing-platform-is-now-available-for-personal-accounts/
It states:
>Existing users on personal accounts will gain access to the enhanced billing platform in the coming months. You will be informed via email and an in-app banner on the billing page in advance of the transition.

So you should be able to use it right now, or in the coming months.

@Railway-Hobby 
I made a VS Code extension to show the remaining hours in the statusbar of VS Code.
Maybe it is usable for your needs. Check it out here: https://marketplace.visualstudio.com/items?itemName=cschot.codespaces-usage-monitor
